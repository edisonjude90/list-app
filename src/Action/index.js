import { SEARCH_PERSON, VIEW_PERSON_DETAILS } from "../Constants";

export const SEARCH_PERSON_ACTION = (payload) => ({
    type:SEARCH_PERSON,
    payload:payload
});

export const VIEW_PERSON_DETAILS_ACTION = (payload) => ({
    type:VIEW_PERSON_DETAILS,
    payload:payload
});
