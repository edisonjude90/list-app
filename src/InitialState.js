const AppState = {
    personList:[
        {
            id: 1,
            first: "John",
            last: "Roberts",
            age: 51,
            dob: "24-05-1967",
            location: "Chicago, IL",
            description: "John is a retired YouTuber"
        },
        {
            id: 2,
            first: "Frita",
            last: "Waters",
            age: 27,
            dob: "24-05-1991",
            location: "Phoenix, AZ",
            description: "Frita loves the desert, cheese, and turtles."
        },
        {
            id: 3,
            first: "Bart",
            last: "Simpson",
            age: 24,
            dob: "24-05-1996",
            location: "Springfield, IL",
            description: "Bart skateboards everywhere."
        },
        {
            id: 4,
            first: "Harry",
            last: "Simpson",
            age: 24,
            dob: "24-05-1996",
            location: "New York, US",
            description: "Likes Nature"
        },
        {
            id: 5,
            first: "David",
            last: "Mark",
            age: 24,
            dob: "24-05-1990",
            location: "Phoenix, AZ",
            description: "David is a DJ"
        },
        {
            id: 6,
            first: "Jason",
            last: "Williams",
            age: 28,
            dob: "10-01-1990",
            location: "New York, US",
            description: "Likes Bowling"
        }
    ],
    searchResult:[],
    viewDetails:{}
};

export default AppState;