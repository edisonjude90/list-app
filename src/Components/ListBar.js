import React from "react";
import { connect } from "react-redux";
import { SEARCH_PERSON_ACTION , VIEW_PERSON_DETAILS_ACTION } from "../Action";

const mapStateToProps = (state) => ({
    searchResult:state.searchResult
});

const mapDispatchToProps = (dispatch)=>{
    return {
        searchPerson:(payload) => (dispatch(SEARCH_PERSON_ACTION(payload))),
        viewDetailsFor:(payload) => (dispatch(VIEW_PERSON_DETAILS_ACTION(payload)))
    };
};

class ConnectListBar extends React.Component {

    constructor(props){
        super(props);
        this.state = {
            searchText:""
        }
        this.handleChange = this.handleChange.bind(this); 
    }

    handleChange(e){
        this.setState({
            searchText:e.target.value
        },()=>{
            this.props.searchPerson({
                name:this.state.searchText
            });
        });
    }

    ViewDetails(id){
        this.props.viewDetailsFor({id:id});
    }
    
    render(){
        
        var allResult = this.props.searchResult.map((item)=>(
            <li key = {item.id} onClick = { ()=>{ this.ViewDetails(item.id) } } >
                <span>{ item.first + " " + item.last }</span>
            </li>
        ));

        if (allResult.length === 0){
            allResult = <li> No Results Found </li>;
        }

        return (
                <div className="list-bar">
                    <header>
                        <h2>List</h2>
                    </header>
                    <div className="search-box">
                        <input type="text" placeholder = "Search" value = { this.state.searchText } onChange = { this.handleChange } />
                    </div>
                    <ul className = "person-list">
                        { allResult }
                    </ul>
                </div>
        );

    }

}

const ListBar = connect(mapStateToProps,mapDispatchToProps)(ConnectListBar);

export default ListBar;