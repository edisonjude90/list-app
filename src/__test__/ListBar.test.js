import React from 'react';
import { createStore } from "redux";
import renderer from 'react-test-renderer';

import InitialState from '../InitialState';

import { SEARCH_PERSON_ACTION , VIEW_PERSON_DETAILS_ACTION } from "../Action";

import reducer from '../Reducer';

import ListBar from '../Components/ListBar';

it('ListBar Component list all users', ()=>{
    const store = createStore(reducer,InitialState);
    const ListBarSection = renderer.create(<ListBar store = { store } />).toJSON();
    expect(ListBarSection).toMatchSnapshot(); 
});

it('ListBar Component list searched user', ()=>{
    const store = createStore(reducer,InitialState);
    store.dispatch(SEARCH_PERSON_ACTION({name:"john"}));
    const ListBarSection = renderer.create(<ListBar store = { store } />).toJSON();
    expect(ListBarSection).toMatchSnapshot();
});

