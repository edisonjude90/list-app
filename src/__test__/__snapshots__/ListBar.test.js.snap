// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBar Component list all users 1`] = `
<div
  className="list-bar"
>
  <header>
    <h2>
      List
    </h2>
  </header>
  <div
    className="search-box"
  >
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  <ul
    className="person-list"
  >
    <li
      onClick={[Function]}
    >
      <span>
        John Roberts
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        Frita Waters
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        Bart Simpson
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        Harry Simpson
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        David Mark
      </span>
    </li>
    <li
      onClick={[Function]}
    >
      <span>
        Jason Williams
      </span>
    </li>
  </ul>
</div>
`;

exports[`ListBar Component list searched user 1`] = `
<div
  className="list-bar"
>
  <header>
    <h2>
      List
    </h2>
  </header>
  <div
    className="search-box"
  >
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  <ul
    className="person-list"
  >
    <li
      onClick={[Function]}
    >
      <span>
        John Roberts
      </span>
    </li>
  </ul>
</div>
`;
